File: compute_science/insertionSort.js
let testArr = [3, 2, 1, 12, 0, -2, 4, 2];

function insertionSort(arr) {
  for (let i = 1; i < arr.length; i++) {
    let currentValue = arr[i];
    let j = i - 1;

    while (arr[j] > currentValue && j >= 0) {
      console.log("arr[j + 1]: ", arr[j + 1], j + 1);
      arr[j + 1] = arr[j];
      j--;
    }
    arr[j + 1] = currentValue;
  }
  return arr;
}

insertionSort(testArr);
console.log("sorted:", testArr);

File: compute_science/insertionSort.py

def insertionSort(arr: list[int]):
  for i in range(1, len(arr)):
    value = arr[i]
    while i > 0 and value < arr[i - 1]:
      arr[i] = arr[i-1]
      i = i - 1
    arr[i] = value
  return arr

print(insertionSort([3,1, 9, 0, 4]))